_posts/2020-10-27-scalajs-for-scala-3.md

```diff
@@ -0,0 +1,184 @@
+---
+layout: blog-detail
+post-type: blog
+by: Sébastien Doeraene
+title: "Implementing Scala.js Support for Scala 3"
+---
+
+Today, the JS platform has become an integral part of the language.
+Yet, until last August, support for Scala.js in Scala 3 was close to non-existent.
+A first preview was shipped in Dotty 0.27.0-RC1, with support for the portable subset of Scala and native JS types.
+Since then, support for non-native JS types was added and will ship as part of Scala 3.0.0-M1.
+The only missing feature left is JS exports, which we will implement by the next milestone.
+
+What did it take to add Scala.js support in Scala 3?
+How come we managed to do the necessary work in less than 3 months, whereas Scala.js for Scala 2 took 7 years to mature?
+Those are questions we aim to explore in this post.
+After some background on the architecture of Scala.js for Scala 2, we will cover which components had to be rewritten for Scala 2, and which ones we could reuse.
+We will see that only a small fraction required a rewrite.
```

Comment:
I feel like the intro could indicate more strongly who will want to read this and who won't. Like is this only of interest to compiler hackers?

As I always say on every blog post PR, I think the intro should give away the conclusions. Don't make the reader read the whole thing to get the basic takeaways — it's not suspense fiction.

Reply:
I had tried to anticipate your comment about that (having seen it on previous blog posts), but clearly what I wrote was not good enough. Is it better now?

Reply:
Yes, I think this was an improvement.